Simplified toolbar functionality

archive64 opened the pull request.

Commit:
remove more buttons

Hide these buttons:
  * Rotate ccw
  * Search

We're not using them in the mockup.

Commit:
show "rotate cw" button in toolbar

In the default viewer, the rotate cw button
was only shown in the dropdown menu.

We want to display it in the main toolbar
on larger screen sizes. Keep it in the dropdown
for smaller screen sizes.

Commit:
hide dropdown menu on larger screen sizes

On larger screen sizes, we aren't displaying
_any_ of the buttons in the dropdown menu.

In that case, let's hide the dropdown menu
button on large screen sizes. When the screen
gets smaller, we move some of the toolbar
buttons into the dropdown, so we can show
the button again.

Commit:
Modify "page 1 of n" display

In the standard viewer, the page number is in
an <input> box, but we want to display it in
a <span>.

Some of the js relies on this <input> box
existing, so we'll hide it and put the new
<span> next to it. A later commit will add
the js to make this functional.

Also, change "Page: " to "page ".

Commit:
swap position of "zoom" and "page up/down" buttons

In the mockup, we want to display the "zoom" buttons
on the left and the "page up/down" buttons in the middle.

Also, hide the dropdown menu for the zoom level, since that
isn't in the mockup.

Commit:
add customize_viewer.js file for custom behavior

I'd like to avoid modifying too many of the files
from this fork, because it will make it more difficult
to update pdf.js versions. So I'm adding a new js file for
any custom functionality.

The gulpfile needs to include this file in the final build.
This could have also been achieved by `require`ing this file
from one of the other files, but I thought this file seemed
more like an "entry point" than a dependency.

Also, add it to `viewer.js` so that it gets loaded when testing
with `gulp server`.

The module header is copied and edited from the other js
files in this repo.

Commit:
Make pageNumberDisplay and rotateCw button work correctly

Since these weren't supported in the original viewer,
add some js to make them functional.

Commit:
watch portrait/landscape viewing orientation

When the user is viewing the pdf in landscape, add
class="landscape" to #mainContainer. We'll use this
to re-style the rotate button, as requested in the
mockup.

Commit:
add customize_viewer.css file for custom styling

I'd rather have our custom styling stay in a
separate file for now, to prevent merge
conflicts when pulling from upstream.

Since this pull request is just for new
functionality, I don't want to include
too much custom css. These are the just the
mimimum styles to make the pdf viewer not
look broken. That way, we can deploy this
as-is without waiting for future pull
requests.

Commit:
commit build files

solflare commented on build/generic/web/viewer.html:
```diff
@@ -186,24 +186,24 @@
-                     </button>
-                  </div>
-                  <span id="scaleSelectContainer" class="dropdownToolbarButton">
+                  <button class="toolbarButton pageUp" title="Previous Page" id="previous" tabindex="13" data-l10n-id="previous">
```
Should we update the tabindices here and on others that have been moved?

archive64 replied:
Yeah, good catch!

Commit:
fix tabindexes

I moved the elements around, but forgot to
reorder the tabindex attributes.

Commit:
commit build files

Commit:
fix icon for retina displays

All of the other icons use the @2x size on retina screens.
We have a general css rule that scales icons on retina
screens by 0.5, so using the regular-sized icon caused
it to be displayed too small.

Commit:
comment out the "transform:rotate" for landscape

We'll wait to do this until we have the correct
icon for the mockup.

Commit:
hide "page" label on small screens

Change "page 1 of n" to "1 of n" on small
screens. This lets the window get slightly
narrower.

Commit:
commit build files

solflare commented on build/generic/web/viewer.html:
```diff
@@ -240,6 +240,7 @@
           <menuitem id="contextPageRotateCw" label="Rotate Clockwise"
```
Do we want this to be hidden as well?

archive64 replied:
I don't think so, since we're still allowing users to use the equivalent button.

Now that I think about it, I'm not sure if `class="hidden"` works on `<menuitem>`s anyway.

kimarre commented on build/generic/web/viewer.css:
```diff
@@ -264,6 +264,64 @@
+}
+
+#mainContainer.landscape .rotateCw::before {
+  /* transform: rotate(-90deg); */
```
Is this commented out on purpose?

archive64 replied:
Yes. We're waiting to apply this effect until we start using our custom icons.

copperwall commented:
CR :fire:
